calico_containers/pycalico/netns.py

```diff
@@ -51,6 +52,55 @@ def setup_logging(logfile, level=logging.INFO):
     _log.addHandler(handler)
 
 
+def increment_metrics(namespace):
+    """
+    Increment the metric of any default route that has a metric of 0.
+    :param namespace: The Networking namespace of the container.
+    :return: None. Raises CalledProcessError on error.
+    """
+    def get_metric(route):
+        """
+        Gets the metric of a route.
+        :param route: String representation of a route (full line returned by
+        `ip route`.
+        :return: Integer value representing the metric of the route.
+        """
+        match = re.search('metric\s+(\d)', route)
+        if match:
+            metric = int(match.group(1))
+        else:
+            metric = 0
+        return metric
+
+    with NamedNamespace(namespace) as ns:
+        routes = ns.check_output(["ip", "route"]).split("\n")
+        default_routes = []
+        route_with_metric_0 = False
+        for route in routes:
+            if route.startswith("default"):
+                if "metric" not in route:
+                    # At least one route
+                    route_with_metric_0 = True
+                default_routes.append(route)
+
+        if route_with_metric_0:
+            # Sort default routes by descending metric
+            descending_routes = sorted(default_routes,
+                                       lambda route_a, route_b:
+                                       cmp(get_metric(route_b),
+                                           get_metric(route_a)))
+            # Increment metric of each route by adding a new route,
+            # then removing the old route.
+            for route in descending_routes:
```

Comment:
That's a good point.  The string you get back is structured data; we could just parse it.  Would also combine with other places in this module where we manipulate the routing table.